### PR TITLE
[IE][CMAKE] Fix for in-tree generated InferenceEngineConfig.cmake

### PR DESCRIPTION
**Pull request by vinograd47**
Add `IE::` prefixed aliases for provided targets.

### CODE REVIEW

**Comment by vinograd47** [comment]
Locally I have an issues on Windows with `template_extension` library due to this. It links `IE::inference_engine` target while only `inference_engine` is defined. Don't know why it works on CI and Linux.

File: inference-engine/cmake/config.cmake.in [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 if(DEFINED IE_MAIN_SOURCE_DIR AND TARGET inference_engine)
     set(InferenceEngine_LIBRARIES inference_engine inference_engine_c_api)
+    add_library(IE::inference_engine ALIAS inference_engine)
+    if(TARGET inference_engine_c_api)
+        add_library(IE::inference_engine_c_api ALIAS inference_engine_c_api)
+    endif()
```
Comment (ilya-lavrenov) [inline inference-engine/cmake/config.cmake.in:10]: should not we set like this?
```
set(InferenceEngine_LIBRARIES IE::inference_engine IE::inference_engine_c_api)
```

**Review by ilya-lavrenov — APPROVED** [review]

File: inference-engine/cmake/config.cmake.in (hunk 1 above)
Comment (vinograd47) [reply inference-engine/cmake/config.cmake.in:10]: In this case, I think, it doesn't matter. Just some 3rd-party projects use `InferenceEngine_LIBRARIES` variable, while others use `IE::inference_engine` directly.

File: inference-engine/cmake/config.cmake.in [hunk 2]
```diff
@@ -4,6 +4,10 @@
 
 if(DEFINED IE_MAIN_SOURCE_DIR AND TARGET inference_engine)
     set(InferenceEngine_LIBRARIES inference_engine inference_engine_c_api)
+    add_library(IE::inference_engine ALIAS inference_engine)
+    if(TARGET inference_engine_c_api)
+        add_library(IE::inference_engine_c_api ALIAS inference_engine_c_api)
```
Comment (ilya-lavrenov) [inline inference-engine/cmake/config.cmake.in:9]: should not we check that target is already defined? E.g.
```cmake
find_package(InferenceEngine)
find_package(InferenceEngine)
```
should work

Comment (vinograd47) [reply inference-engine/cmake/config.cmake.in:11]: Done

**Review by ilyachur — APPROVED** [review]

**Pull request merged**